File: maths/Codes/Diagonals Length in a Regular Polygon.cpp
#include <iostream>
#include <math.h>

int get_integer(string msg = "Number: ") {
    int i;
    cout << msg;
    cin >> i;
    return i;
}

double get_float(string msg = "Number: ") {
    double f;
    cout << msg;
    cin >> f;
    return f;
}

using namespace std;

int main() {
    while (true) {
        int n = get_integer("No of sides: ");
        double a = get_float("Each side length: ");
        float diagonals[n - 2];
        diagonals[0] = a;
        double d = n - 3;
        int temp = floor(d / 2);
        double value;

        for (int i = 0; i < d; i++) {
            if ((i + 1) <= ceil(d / 2)) {
                value = diagonals[i] * cos(M_PI / n) + (diagonals[0] * cos((i + 1) * M_PI / n));
            }
            else {
                value = diagonals[temp];
                temp -= 1;
            }
            diagonals[i+1] = value;
        }

        for (int i  = 1; i < n - 2; i++) {
            cout << "Diagonal " << i << " = " << diagonals[i] << endl;
        }
        cout << endl;
    }
}

File: maths/codes/Diagonals Length in a Regular Polygon.py
from math import ceil, cos, pi

def get_integer(msg="Number: ", wrng_msg="Invalid Input"):
    try:
        i = int(input(msg))
    except ValueError:
        print(wrng_msg)
        return get_integer(msg, wrng_msg)
    else:
        return i


def get_float(msg="Number: ", wrng_msg="Invalid Input"):
    try:
        f = float(input(msg))
    except ValueError:
        print(wrng_msg)
        return get_float(msg, wrng_msg)
    else:
        return f


while True:
    n = get_integer("No of sides: ")
    a = get_float("Each side length: ")
    diagonals = [a]
    d = n - 3

    if d % 2 == 0:
        temp = -1
    else:
        temp = -2

    for i in range(d):
        if i + 1 <= ceil(d / 2):
            value = (diagonals[-1] * cos(pi / n)) + (diagonals[0] * cos((i + 1) * pi / n))
        else:
            value = diagonals[temp]
            temp -= 2
        diagonals.append(round(value, 3))

    for i in range(1, len(diagonals)):
        print(f"Diagonal {i} = {diagonals[i]}")
    print()





























